### PR TITLE
Adding Fluid Dynamics to RomApp

### PR DESCRIPTION
**Pull request by Rbravo555**
This PR allows the RomApplication to run reduced order models of fluid dynamics cases using the monolithic solver of the FluidDynamicsApplication

The test files retrieve the geometry directly from the CouetteFlowTest of the FluidDynamicsApplication

### CODE REVIEW

File: applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        ## Solver construction
+        with open('RomParameters.json') as rom_parameters:
+            rom_settings = KratosMultiphysics.Parameters(rom_parameters.read())
+            self.project_parameters["solver_settings"].AddValue("rom_settings", rom_settings["rom_settings"])
+        return solver_wrapper.CreateSolverByParameters(self.model, self.project_parameters["solver_settings"],self.project_parameters["problem_data"]["parallel_type"].GetString())
+
+    def _GetSimulationName(self):
+        return "::[ROM Simulation]:: "
+
+    def ModifyInitialGeometry(self):
```
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py]: It is a bit counterintuitive to me to do this in the `ModifyInitialGeometry` function. Could it be done in `ModifyAfterSolverInitialize`? (not sure if you need your solver to have this values set before calling the `Initialize`. If this is the case, I think we could propose introducing a ` ModifyBeforeSolverInitialize` to the `AnalysisStage`.

File: applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        with open('RomParameters.json') as f:
+            data = json.load(f)
+            nodal_dofs = len(data["rom_settings"]["nodal_unknowns"])
+            nodal_modes = data["nodal_modes"]
+            counter = 0
+            rom_dofs= self.project_parameters["solver_settings"]["rom_settings"]["number_of_rom_dofs"].GetInt()
+            for node in computing_model_part.Nodes:
+                aux = KratosMultiphysics.Matrix(nodal_dofs, rom_dofs)
+                for j in range(nodal_dofs):
+                    Counter=str(node.Id)
```
Comment (rubenzorrilla) [inline applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py:39]: Why casting to string?

File: applications/RomApplication/tests/fluid_dynamics_test_files/MainKratosROM.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+import KratosMultiphysics
+from KratosMultiphysics.RomApplication.fluid_dynamics_analysis_rom import FluidDynamicsAnalysisROM
+
+import sys
+import time
+import numpy as np
+import pdb
```
Comment (rubenzorrilla) [inline applications/RomApplication/tests/fluid_dynamics_test_files/MainKratosROM.py]: I think you're not using this

File: applications/RomApplication/tests/fluid_dynamics_test_files/test_ROM.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+from .MainKratosROM import TestFluidDynamicsROM
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import KratosMultiphysics.kratos_utilities as kratos_utilities
+
+
+
+class ROMFluidDynamics(KratosUnittest.TestCase):
+#########################################################################################
+
+    def test_Fluid_Dynamic_ROM_2D(self):
```
Comment (rubenzorrilla) [inline applications/RomApplication/tests/fluid_dynamics_test_files/test_ROM.py]:
```suggestion
    def test_Fluid_Dynamics_ROM_2D(self):
```

File: applications/RomApplication/tests/fluid_dynamics_test_files/test_ROM.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            ObtainedOutput = Simulation.EvaluateQuantityOfInterest()
+            ExpectedOutput = np.load('ExpectedOutput.npy')
+
+            for i in range (np.shape(ObtainedOutput)[1]):
+                UP=0
+                DOWN=0
+                UP = sum((ExpectedOutput[:,i] - ObtainedOutput[:,i])**2)
+                DOWN = sum((ExpectedOutput[:,i])**2)
+                L2 = np.sqrt(UP/DOWN)
+                self.assertLess(L2, 1e-12)
```
Comment (rubenzorrilla) [inline applications/RomApplication/tests/fluid_dynamics_test_files/test_ROM.py]:
```suggestion
                up=0
                down=0
                up = sum((ExpectedOutput[:,i] - ObtainedOutput[:,i])**2)
                down = sum((ExpectedOutput[:,i])**2)
                l2 = np.sqrt(up/down)
                self.assertLess(l2, 1e-12)
```

Comment (rubenzorrilla) [reply applications/RomApplication/tests/fluid_dynamics_test_files/test_ROM.py]: Besides the style guide issue, I think you don't need to initialize up and down to zero.

File: applications/RomApplication/tests/test_RomApplication.py [hunk 6]
```diff
@@ -37,6 +38,7 @@ def AssembleTestSuites():
     smallSuite.addTest(ROMStaticStruct('test_Struct_Static_ROM_2D'))
     smallSuite.addTest(ROMStaticStruct('test_Struct_Static_HROM_2D'))
     smallSuite.addTest(ROMDynamicStruct('test_Struct_Dynamic_ROM_2D'))
+    smallSuite.addTest(ROMFluidDynamics('test_Fluid_Dynamic_ROM_2D'))
```
Comment (rubenzorrilla) [inline applications/RomApplication/tests/test_RomApplication.py]:
```suggestion
    smallSuite.addTest(ROMFluidDynamics('test_Fluid_Dynamics_ROM_2D'))
```

File: applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py (hunk 2 above)
Comment (Rbravo555) [reply applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py:39]: We are calling the key in the JSON file in which the nodal modes are contained, that's why it is a string...

File: applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py (hunk 1 above)
Comment (Rbravo555) [reply applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py]: I have no objection. I require the nodes to have their ROM_BASIS and the Elements to have their ROM_WEIGHT before the SetSetUpDofSet method of the RomBuilder&Solver is called. I changed it as suggested

File: applications/RomApplication/tests/fluid_dynamics_test_files/MainKratosROM.py (hunk 3 above)
Comment (Rbravo555) [reply applications/RomApplication/tests/fluid_dynamics_test_files/MainKratosROM.py]: addressed

File: applications/RomApplication/tests/fluid_dynamics_test_files/test_ROM.py (hunk 4 above)
Comment (Rbravo555) [reply applications/RomApplication/tests/fluid_dynamics_test_files/test_ROM.py]: changed

File: applications/RomApplication/tests/fluid_dynamics_test_files/test_ROM.py (hunk 5 above)
Comment (Rbravo555) [reply applications/RomApplication/tests/fluid_dynamics_test_files/test_ROM.py]: changed

File: applications/RomApplication/tests/test_RomApplication.py (hunk 6 above)
Comment (Rbravo555) [reply applications/RomApplication/tests/test_RomApplication.py]: changed

File: applications/RomApplication/tests/structural_static_test_files/MainKratosHROM.py [hunk 7]
```diff
@@ -8,10 +6,10 @@
 class TestStructuralMechanicsStaticHROM(StructuralMechanicsAnalysisROM):
 
     def __init__(self,model,project_parameters):
-        super(TestStructuralMechanicsStaticHROM,self).__init__(model,project_parameters)
+        super().__init__(model,project_parameters)
 
     def ModifyInitialGeometry(self):
-        super(TestStructuralMechanicsStaticHROM,self).ModifyInitialGeometry()
+        super().ModifyInitialGeometry()
```
Comment (rubenzorrilla) [inline applications/RomApplication/tests/structural_static_test_files/MainKratosHROM.py]: Shouldn't this be `ModifyAfterSolverInitialize` after what we spoke before?

File: applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py (hunk 1 above)
Comment (rubenzorrilla) [reply applications/RomApplication/python_scripts/fluid_dynamics_analysis_rom.py]: I think this is a better usage of the `AnalysisStage` methods. Thanks!

File: applications/RomApplication/tests/structural_static_test_files/MainKratosHROM.py (hunk 7 above)
Comment (Rbravo555) [reply applications/RomApplication/tests/structural_static_test_files/MainKratosHROM.py]: Indeed, I have changed it as well now

**Review by rubenzorrilla — APPROVED** [review]
For me this is ready to go in.

**Comment by rubenzorrilla** [comment]
Now I think we should add you application to the CI.

**Pull request merged**